receiver/splunkhecreceiver/receiver.go

```diff
@@ -391,9 +392,20 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 	}
 }
 
-func (r *splunkReceiver) extractChannelHeader(req *http.Request) (string, bool) {
-	if headers, ok := req.Header[splunk.HTTPSplunkChannelHeader]; ok {
```

Comment:
Why not just use `req.Header.Get`? It is case insensitive

Reply:
The reason why we ended up not using req.Header.Get is that it returns empty string on both the case where the header does not exists and when the value assigned to the header is empty. So we will not be able to distinguish the two cases.